packages/nextjs/components/Search.stories.tsx

```diff
@@ -0,0 +1,126 @@
+import type { Meta, StoryObj } from ".storybook/types";
+import { expect } from "@storybook/jest";
+import { userEvent, waitFor, within } from "@storybook/testing-library";
+
+import { Search } from "./Search";
+
+const meta: Meta<typeof Search> = {
+  component: Search,
+  args: {},
+};
+
+export default meta;
+
+type Story = StoryObj<typeof Search>;
+
+export const Default: Story = {
+  async play({ canvasElement, step }) {
+    const ui = wrap(canvasElement);
+    await step("expect to see a search box", async () => {
+      expect(ui.searchbox).toBeVisible();
+    });
+  },
+};
+
+export const WithSearchTerm: Story = {
+  async play({ canvasElement, step }) {
+    const ui = wrap(canvasElement);
+
+    await step("Type 'baguette' into the search box", async () => {
+      ui.searchbox.focus();
+      await userEvent.type(ui.searchbox, "baguette");
+    });
+
+    await step("expect to see a loading indicator", async () => {
+      expect(ui.resultsBox).toBeVisible();
+      expect(ui.resultsText).toEqual(["Loading..."]);
+    });
+
+    await step("expect to see some search results", async () => {
+      await waitFor(
+        () => {
+          expect(ui.resultsText).toHaveLength(2);
+        },
+        { timeout: 3000 }
+      );
+    });
+  },
+};
+
+export const WithSearchTerm_Test_FilteredResults: Story = {
+  name: "With Search Term / Test / Filtered Results",
+  async play(ctx) {
+    // eslint-disable-next-line storybook/context-in-play-function
+    await WithSearchTerm.play!(ctx);
+
+    const { canvasElement, step } = ctx;
+    const ui = wrap(canvasElement);
+
+    await step("type 'plain'", async () => {
+      await userEvent.type(ui.searchbox, " plain");
+    });
+
+    await step("there should be one result", async () => {
+      await waitFor(() => {
+        expect(ui.resultsText).toEqual(["Plain Baguette"]);
+      });
+    });
+  },
+};
+export const WithSearchTerm_Test_NoResults: Story = {
+  name: "With Search Term / Test / No Results",
+  async play(ctx) {
+    // eslint-disable-next-line storybook/context-in-play-function
+    await WithSearchTerm.play!(ctx);
+
+    const { canvasElement, step } = ctx;
+    const ui = wrap(canvasElement);
+
+    await step("update text to 'cake'", async () => {
+      await userEvent.clear(ui.searchbox);
+      await userEvent.type(ui.searchbox, "cake");
+    });
+
+    await step("there should be no results", async () => {
+      await waitFor(() => {
+        expect(ui.resultsText).toEqual(["No Products Found"]);
+      });
+    });
```

Comment:
I see a bit of flakiness here.
![image](https://github.com/FormidableLabs/nextjs-sanity-fe/assets/588535/fed2a3a7-3837-48ff-a1b6-65345fd8898b)
![image](https://github.com/FormidableLabs/nextjs-sanity-fe/assets/588535/4eb10e0f-a3d1-4637-bd4f-89946087e273)

Reply:
(as discussed) this is a fluke with the current Search box ... it requires the "focus" to be kept inside the search box, otherwise the search results are hidden.  
So if you're clicking around while the test is running, it will interfere.